src/datasources/floodscan.py

```diff
@@ -0,0 +1,82 @@
+import os
+from pathlib import Path
+
+import rioxarray as rxr
+import xarray as xr
+
+from src.utils import cloud_utils, cog_utils, gen_utils
+
+# from src.utils import blob
+
+DATA_DIR_GDRIVE = Path(os.getenv("AA_DATA_DIR_NEW"))
+FP_FS_HISTORICAL = (
+    DATA_DIR_GDRIVE
+    / "private"
+    / "raw"
+    / "glb"
+    / "FloodScan"
+    / "SFED"
+    / "SFED_historical"
+    / "aer_sfed_area_300s_19980112_20231231_v05r01.nc"
+)
+
+
+def open_historical_floodscan():
+    chunks = {"lat": 1080, "lon": 1080, "time": 1}
+    ds = xr.open_dataset(FP_FS_HISTORICAL, chunks=chunks)
+    da = ds["SFED_AREA"]
+    da = da.rio.set_spatial_dims(x_dim="lon", y_dim="lat")
+    da = da.rio.write_crs(4326)
+    return da
+
+
+def load_floodscan_cogs(
+    start_date,
+    end_date,
+    mode="dev",
+    container_name="global",
+    prefix="raster/cogs/aer_area_300s",
+):
+    container_client = cloud_utils.get_container_client(
+        mode=mode, container_name=container_name
+    )
+    cogs_list = [
+        x.name
+        for x in container_client.list_blobs(name_starts_with=prefix)
+        if (gen_utils.extract_date(x.name).date() >= start_date)
+        & (gen_utils.extract_date(x.name).date() <= end_date)  # noqa
+    ]
+
+    das = []
+    for cog in cogs_list:
+        da_in = process_floodscan_cog(
+            cog_name=cog, container_name="global", mode="dev"
+        )
+        das.append(da_in)
+    return xr.combine_by_coords(das, combine_attrs="drop")
```

Comment:
If just stacking by a single dimension at once, can use `xr.concat()` instead (and then you don't need to do `da_in = da_in.expand_dims()` beforehand). But perhaps this is better anyways because I think it looks for duplicate coords.